ext/storage_sources/s3_store/README.md

```diff
@@ -0,0 +1,77 @@
+# WiredTiger's S3 Store Extension
```

Comment:
Maybe just say "S3 Extension"? That is how we refer to it throughout the remainder of the document.